daft/io/_sql.py

```diff
@@ -10,30 +10,67 @@
 from daft.logical.builder import LogicalPlanBuilder
 from daft.sql.sql_scan import SQLScanOperator
 
+if TYPE_CHECKING:
+    from sqlalchemy.engine import Connection
+
 
 @PublicAPI
 def read_sql(
-    sql: str, url: str, partition_col: Optional[str] = None, num_partitions: Optional[int] = None
+    sql: str,
+    conn: Union[Callable[[], "Connection"], str],
```

Comment:
Does `"Connection"` need to be in a string? Or can we use the `Connection` class that's been imported

Reply:
I think it does cuz the Connection class is under a `if TYPE_CHECKING:` rule, similar to other methods like from_arrow `def from_arrow(data: Union["pa.Table", List["pa.Table"]]) -> "DataFrame":`

Also if I unquoted it the precommit checks will complain